### ISSUE TITLE
rename `beforeAll/afterAll` to `pre/post`?

### ISSUE THREAD

**Issue by ianstormtaylor** [egoist/maid#7]
Hey, thanks for the interesting tool!

I was just gonna make a suggestion. Right now you can have things like `postbuild` and `prebuild` which run automatically. But the automatic before/after everything is confusingly named `afterAll` and `beforeAll`, which isn't consistent.

It might be nice to allow `pre` and `post` tasks that run before/after everything instead, to stay consistent. It feels natural that `pre` without `pre{task}` would count for everything.

**Comment by ianstormtaylor** [comment]
@zephraph I'm not sure I totally understand. I think "before anything else runs" and "pre all tasks" are kind of synonymous. I'd personally avoid any of the plain english stuff, since I think it makes the API really hard to remember.

**Comment by chrisdmacrae** [comment]
I disagree @ianstormtaylor. It should be plain English. That's what makes this so interesting. 

However it should still be a standard API. 

`<action> <condition> <target> [target identifier]`

In our case, these are:

`run [before|after] [task|all]`

Or `run before all`, `run after all`

**Comment by chrisdmacrae** [comment]
Upon thinking about the above more, it wont work.

How do I run multiple "before all tasks" before themselves?

Having a single command called "beforeAll", "afterAll" makes sense, because there should only be one or the other.

**Comment by zephraph** [comment]
You could still do a label, just error out if there's more than one. I think the current beforeAll, afterAll semantics are fine too though.
